controllers/nutanixcluster_controller.go

```diff
@@ -178,12 +182,21 @@ func (r *NutanixClusterReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 	}
 	conditions.MarkTrue(cluster, infrav1.CredentialRefSecretOwnerSetCondition)
 
-	v3Client, err := CreateNutanixClient(ctx, r.SecretInformer, r.ConfigMapInformer, cluster)
+	v3Client, err := prismclientcache.DefaultCache.Get(cluster.Name)
 	if err != nil {
-		conditions.MarkFalse(cluster, infrav1.PrismCentralClientCondition, infrav1.PrismCentralClientInitializationFailed, capiv1.ConditionSeverityError, err.Error())
-		return ctrl.Result{Requeue: true}, fmt.Errorf("nutanix client error: %v", err)
+		if errors.Is(err, prismclientcache.ErrorClientNotFound) {
+			log.Info("Nutanix client not found in cache. Creating new client")
+			v3Client, err = CreateNutanixClient(ctx, r.SecretInformer, r.ConfigMapInformer, cluster)
+			if err != nil {
+				conditions.MarkFalse(cluster, infrav1.PrismCentralClientCondition, infrav1.PrismCentralClientInitializationFailed, capiv1.ConditionSeverityError, err.Error())
+				return ctrl.Result{Requeue: true}, fmt.Errorf("nutanix client error: %v", err)
```

Comment:
we should add a time interval of how often should k8s control loop call back due to error.

Reply:
That's a good point, an exponential backoff might make more sense (albeit until a certain ceiling). Although, I'd stay away from making a change about it in this PR to keep the scope limited to introducing a cache.

Reply:
this will be needed sooner than later to exponentially backoff given if user gives a wrong password then we will lock the pc due to too many calls with wrong creds

Reply:
fair enough. Updated the `Reconcile: true` calls to `ReconcileAfter: fiveSeconds`